lib/Sema/TypeCheckStmt.cpp

```diff
@@ -508,7 +508,8 @@ class StmtChecker : public StmtVisitor<StmtChecker, Stmt*> {
       Type contextType = yieldType;
       if (yieldResults[i].isInOut()) {
         contextTypePurpose = CTP_YieldByReference;
-        contextType = LValueType::get(contextType);
+        if (!contextType->hasError())
```

Comment:
This check should be before this `if (yieldResults[i].isInOut())` and `!` should be removed from it, also it might be good to check for empty type as well e.g. `(!yieldType || yieldType->hasError())`